module/utils/actor-generator.js

```diff
@@ -113,8 +144,8 @@ export class ActorGenerator {
         await GetOrCacheAbilities(); await GetOrCacheMoves(); await GetOrCacheCapabilities();
 
         if(options.exists) return ag;
-        
-        await ag.Generate().PrepareArt(options.imgpath ?? undefined);
+
+        await ag.Generate("weighted", undefined, undefined, undefined, stat_randomness, is_shiny, prevent_evolution).PrepareArt(options.imgpath ?? undefined, undefined, is_shiny);
```

Comment:
Okay this is definitely gonna need refactoring. No way I'm allowing triple undefineds to plague the codebase 😂 

Will need to change `Generate` to take in a destructured list of parameters instead, will set this up when I actually do testing of this codebase.

Reply:
Yeah, sorry about that - I was sort of snaking values through there in an effort to avoid changing anything else that might call that function, but a refactor would definitely be cleaner at this point.